Roles/Impostor/DollMaster.cs

```diff
@@ -0,0 +1,317 @@
+                }
+                // Unpossessed after waiting for DollMaster.
+                if (Main.AllPlayerControls[DollMasterTarget.PlayerId].CanBeTeleported())
+                {
+                    _ = new LateTask(() =>
+                    {
+                        if (!WaitToUnPossess) return;
+                        UnPossess(Main.AllPlayerControls[DollMasterTarget.PlayerId], Main.AllPlayerControls[ControllingPlayerId]);
+                        GetPlayersPositions(Main.AllPlayerControls[DollMasterTarget.PlayerId]);
+                        SwapPlayersPositions(Main.AllPlayerControls[DollMasterTarget.PlayerId]);
+                    }, 0.15f, "UnPossess");
+                }
+            }
+        }
+    }
+
+    // Get vent Id that the player is in.
+    private static int GetPlayerVentId(PlayerControl pc)
+    {
+        if (!(ShipStatus.Instance.Systems.TryGetValue(SystemTypes.Ventilation, out var systemType) &&
+              systemType.TryCast<VentilationSystem>() is VentilationSystem ventilationSystem))
+            return 0;
+
+        return ventilationSystem.PlayersInsideVents.TryGetValue(pc.PlayerId, out var playerIdVentId) ? playerIdVentId : 0;
+    }
+
+    // Prepare for a meeting if possessing.
+    public override void OnReportDeadBody(PlayerControl pc, PlayerControl target) // Fix crap when meeting gets called.
+    {
+        if (IsControllingPlayer && controllingTarget != null)
+        {
+            bool shouldAnimate = false;
+            Main.AllPlayerControls[DollMasterPlayerId].RpcShapeshift(Main.AllPlayerControls[DollMasterPlayerId], shouldAnimate);
+            Main.AllPlayerControls[ControllingPlayerId].RpcShapeshift(Main.AllPlayerControls[ControllingPlayerId], shouldAnimate);
+            UnPossess(Main.AllPlayerControls[DollMasterPlayerId], Main.AllPlayerControls[ControllingPlayerId]);
+            Main.AllPlayerSpeed[controllingTarget.PlayerId] = originalSpeed;
+            ReportDeadBodyPatch.CanReport[controllingTarget.PlayerId] = true;
+            ReducedVisionPlayers.Clear();
+        }
+    }
+
+
+    // Prevent possessed player from reporting body.
+    public override bool OnCheckReportDeadBody(PlayerControl reporter, GameData.PlayerInfo deadBody, PlayerControl killer)
+    {
+        if (ReducedVisionPlayers.Contains(reporter.PlayerId)) return false;
+        return true;
+    }
+
+    public override bool CheckMurderOnOthersTarget(PlayerControl killer, PlayerControl target) // Swap player kill interactions to each other when possessing.
+    {
+        if (!IsControllingPlayer || controllingTarget == null || DollMasterTarget == null) return false;
+
+        // If Target as DollMaster Main Body gets killed, kill DollMaster instead.
+        if (target.PlayerId == ControllingPlayerId)
+        {
+            PlayerControl playertarget = PlayerControl.AllPlayerControls[ControllingPlayerId];
+            PlayerControl dollmaster = PlayerControl.AllPlayerControls[DollMasterPlayerId];

```

Comment:
should be put outside of the if statement for simplicity, and use Utils.GetPlayerById(id); / 
 currently you're accessing the # index of the array and not the player adjsoned to the selected id. (to put it simply, if a player leaves, this code is broken lol)

Reply:
(also check if the playercontrols aren't null before trying to run this method)

Reply:
I have adjusted to these changes, thanks for the tip, and also trust me I know it's important to check if the target is Null or not because that was causing me me so many errors and crashes before then lol.